pkg/guacanalytics/patchPlanning.go

```diff
@@ -158,35 +180,52 @@ func exploreIsDependencyFromDepPkg(ctx context.Context, gqlclient graphql.Client
 		return nil
 	}
 
-	dfsNVersion, seenVersion := q.nodeMap[isDependency.Package.Namespaces[0].Names[0].Versions[0].Id]
-	dfsNName, seenName := q.nodeMap[isDependency.Package.Namespaces[0].Names[0].Id]
+	addNodeToQueue(q, q.now, q.nowNode.Depth, PackageVersion, false, nil, isDependency.Package.Namespaces[0].Names[0].Versions[0].Id)
+	addNodeToQueue(q, q.now, q.nowNode.Depth, PackageName, false, []string{isDependency.Package.Namespaces[0].Names[0].Versions[0].Version}, isDependency.Package.Namespaces[0].Names[0].Id)
 
-	if !seenVersion {
-		dfsNVersion = DfsNode{
-			Parent:   q.now,
-			Depth:    q.nowNode.Depth + 1,
-			NodeType: "packageVersion",
-		}
-		q.nodeMap[isDependency.Package.Namespaces[0].Names[0].Versions[0].Id] = dfsNVersion
+	return nil
+}
+
+func exploreIsOccurrenceFromSubject(ctx context.Context, gqlclient graphql.Client, q *queueValues, isOccurrence model.NeighborsNeighborsIsOccurrence) {
+	addNodeToQueue(q, q.now, q.nowNode.Depth, Artifact, false, nil, isOccurrence.Artifact.Id)
+}
+
+func exploreHasSLSAFromArtifact(ctx context.Context, gqlclient graphql.Client, q *queueValues, hasSLSA model.NeighborsNeighborsHasSLSA) {
+	// Check that the subject is not the node inputted itself and being re-added to the queue unnecessarily
+	if q.now != hasSLSA.Subject.Id {
+		addNodeToQueue(q, q.now, q.nowNode.Depth, Artifact, true, nil, hasSLSA.Subject.Id)
 	}
+}
 
-	if !seenName {
-		dfsNName = DfsNode{
-			Parent:       q.now,
-			Depth:        q.nowNode.Depth + 1,
-			NodeType:     "packageName",
-			nodeVersions: []string{isDependency.Package.Namespaces[0].Names[0].Versions[0].Version},
+func exploreIsOccurrenceFromArtifact(ctx context.Context, gqlclient graphql.Client, q *queueValues, isOccurrence model.NeighborsNeighborsIsOccurrence) {
+	// TODO: take into account PkgEqual case where there is another alias of the artifact package that we don't know about
+	// Right now we only explore isOccurrences if their artifact is from a HasSLSA
+	if q.nowNode.slsaSubject {
```

Comment:
hmm, why is `slsaSubject` used here to detemine if the edge is explored?

Reply:
slsaSubject is the field I added to denote that an Artifact is linked to the subject of a HasSLSA.

For the moment, the functionality is to only explore an isOccurrence from an artifact if the Artifact is linked to a HasSLSA because then we are exploring packaged and paths no necissarily affected. However, it depends what we want the desired behavior to be if this should be taken into account.  This TODO is to take into account cases that may be affected by this nuance: // TODO: take into account PkgEqual case where there is another alias of the artifact package that we don't know about.

Reply:
I don't believe this to be true right now. Can you help me understand where this is happening?

"For the moment, the functionality is to only explore an isOccurrence from an artifact if the Artifact is linked to a HasSLSA because then we are exploring packaged and paths no necissarily affected. "

Reply:
in the diagram that showed me earlier *

Reply:
by assumption all nodes that are explored from are nodes of interest, which should be the invariant of things in the queue. Would being a slsaSubject change this?

Reply:
![image](https://github.com/guacsec/guac/assets/49292050/16a54545-2ab7-42c6-beae-69274008e689)
For this diagram, pkg4 will not be added to the queue to be explored because the artifact that the isOccurrence is linked to isn't the subject of a HasSLSA.  

This is what I would think should be desired behavior (expect for the one case I have in the TODO) because in this diagram is Artifact1 is produced from Pkg1, it isn't relevent for the search the fact that Artifact1 is also produced by Pkg4 because that wouldn't mean Pkg4 is also affected by the vulnerability too.

Reply:
Resolved (for this example, pkg4 should be explored because essentially pkg1 and pkg4 are equal and thus is pkg1 is vulnerable so is pkg4).